src/services/parser/message/update-fragment-for-sending.js

```diff
@@ -13,36 +11,23 @@ export async function updateFragmentForSending(
   const sendingAsid = config.deductionsAsid;
   const sendingOdsCode = config.deductionsOdsCode;
 
-  const payloadXml = await new xml2js.Parser({ explicitArray: false }).parseStringPromise(
-    fragmentPayload
-  );
+  const payloadXml = xmlStringToJsObject(fragmentPayload);
+
   const copcUK01 = payloadXml.COPC_IN000001UK01;
-  copcUK01.communicationFunctionRcv.device = updateId(
-    copcUK01.communicationFunctionRcv.device,
-    recipientAsid
-  );
-  copcUK01.communicationFunctionSnd.device = updateId(
-    copcUK01.communicationFunctionSnd.device,
-    sendingAsid
-  );
-  copcUK01.id['$'].root = messageId;
+  updateIdExtension(copcUK01.communicationFunctionRcv.device, recipientAsid);
+  updateIdExtension(copcUK01.communicationFunctionSnd.device, sendingAsid);
+  updateIdRoot(copcUK01, messageId);
 
-  copcUK01.ControlActEvent.author1.AgentSystemSDS.agentSystemSDS.id['$'].extension = recipientAsid;
+  updateIdExtension(copcUK01.ControlActEvent.author1.AgentSystemSDS.agentSystemSDS, recipientAsid);
 
   let payloadInformation = copcUK01.ControlActEvent.subject.PayloadInformation;
-  payloadInformation.id['$'].root = messageId;
-  payloadInformation.pertinentInformation.pertinentPayloadBody.id['$'].root = messageId;
+  updateIdRoot(payloadInformation, messageId)
```

Comment:
Can we add a semicolon ; here please?

```suggestion
  updateIdRoot(payloadInformation, messageId);
```